### ISSUE TITLE
cucumber / gherkin editor for non-java

### ISSUE THREAD

**Issue by han5** [cucumber/cucumber-eclipse#164]
It seems that the editor for cucumber / gherkin feature files does not need to be so java specific.

Personally it looks to be a narrow direction the plugin is going. 

We realize this is likely cucumber-jvm implementation, but the editor could expand beyond that if it supported alternative languages like ruby.

Lately the updates have made it ugly for ruby as there are tons of warnings everywhere b/c there's no java step definition found.  And of course its just become too java oriented.

At least provide a way to turn off the warning flags .. 

Thanks

**Comment by pranasblk** [comment]
+1. For now needed to install it for Cucumber JVM Groovy based due to the warnings.

**Comment by PyvesB** [comment]
Indeed, at least an option to remove the warnings and ideally a full Ruby step definition matching would be nice.

**Comment by Lego3** [comment]
The same is true when working with Python. I haven't been able to find any plugin yet which supports steps written in Python, which is a pity.

**Comment by ilanpillemer** [comment]
Send a pull request that makes it optional and I will merge it in.

**Comment by Lego3** [comment]
Created #181 to turn off matching. Also looking at Python step matching using Pydev (when tests are run with Behave), but I'm not sure if that fits in the plugin as such.

**Comment by PyvesB** [comment]
@qvdk how does this new feature work? My step implementations are written in Ruby, but I am not able to access them from feature files using CTRL+click. Note that I have installed the latest snapshot of the plugin.

**Comment by qvdk** [comment]
Helle @PyvesB 

This is not a new feature for now. The plugin was fixed to be able to accept new implementations of the step definitions scan. 
But currently, we only have an implementation to scan Java language. 

So, for now you can only have the gherkin editor for Ruby projects, but without the glue detection.

Any contribution is welcome, if you would like to add a new implementation for Ruby.

**Comment by PyvesB** [comment]
It would probably be beneficial to update the [contributing documentation](https://github.com/cucumber/cucumber-eclipse/wiki/Start-contributing) and/or fix the project setup, because just importing the project as an existing Maven project as suggested in the Wiki leads to something that doesn't even have the Java nature and that can't be run as an Eclipse application (which is generally how you launch Eclipse plugins for local testing in the Eclipse RCP and RAP Developers package).

**Comment by qvdk** [comment]
I followed this guide and it works for me.
Does you have eclipse PDE plugin installed previously ?

**Comment by PyvesB** [comment]
Yes, I do have the PDE plugin installed.

**Comment by csdano2** [comment]
As I see working with Behave (PyDev, Python) is not supported by this plugin. Am I right or do you see possibility to use this plugin fully with behave (e.g.: with code completion, matching python glue code)?

**Comment by laeubi** [comment]
This is currently not supported there are two options:

1. Provide an initial implementation for python parsing so we can integrate this in the plugin
2. try to pay people so they do it for you

**Comment by csdano2** [comment]
Dear Christoph,

Thank you for your quick answer! It was very important for me!

Regards,
Dániel


Mobilról küldve
